File: src/main/java/pl/training/workshop/m3/s10_isp/step2/CashDesk.java
package pl.training.workshop.m3.s10_isp.step2;

/** Kasa zależy tylko od roli TicketSales. */
public final class CashDesk {
    private final TicketSales backOffice;

    public CashDesk(TicketSales backOffice) {
        this.backOffice = backOffice;
    }

    public String sell(String title, int seat) {
        return "bilet " + backOffice.sellTicket(title, seat) + ": " + title + ", miejsce " + seat;
    }

    public String refund(String ticketId) {
        return backOffice.refundTicket(ticketId);
    }
}

File: csharp/src/Training.Workshop/M3/S10Isp/Step2/CashDesk.cs
namespace Training.Workshop.M3.S10Isp.Step2;

/// <summary>Kasa zależy tylko od roli ITicketSales.</summary>
public sealed class CashDesk
{
    private readonly ITicketSales _backOffice;

    public CashDesk(ITicketSales backOffice)
    {
        _backOffice = backOffice;
    }

    public string Sell(string title, int seat)
    {
        return "bilet " + _backOffice.SellTicket(title, seat) + ": " + title + ", miejsce " + seat;
    }

    public string Refund(string ticketId)
    {
        return _backOffice.RefundTicket(ticketId);
    }
}
